### PR TITLE
Feat/send massage reactive archived thread

### PR DESCRIPTION
**Pull request by giachuvan**

### CODE REVIEW

File: crates/mezon-store/src/channel.rs [hunk 1]
```diff
@@ -2172,6 +2490,7 @@ fn channel_from_desc(
         voice_members,
         is_favorite,
         creator_id: UserId(c.creator_id),
+        active: CHANNEL_ACTIVE_JOINED,
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: Hardcoded to JOINED. ApiChannelDesc drops proto active (field 17), so Channel.active is always 1 after a reload.

File: crates/mezon-ui/src/sidebar/channel_sidebar/mod.rs [hunk 2]
```diff
@@ -270,7 +270,11 @@ impl ChannelSidebar {
                         collapsed,
                     });
                     if !collapsed {
-                        let ch_slice = &category.channels;
+                        let ch_slice: Vec<_> = category
+                            .channels
+                            .iter()
+                            .filter(|ch| ch.visible_in_sidebar())
+                            .collect();
```
Comment (huylexuan44) [inline crates/mezon-ui/src/sidebar/channel_sidebar/mod.rs:339]: The collapsed-category branch at line 346 is not filtered, so archived threads reappear when the category is collapsed.

File: crates/mezon-store/src/channel.rs [hunk 3]
```diff
@@ -1992,6 +2058,231 @@ impl ChannelList {
...
+                for cats in self.cache.values_mut() {
+                    removed |= remove_channel(cats, channel_id);
+                }
+            }
+            if removed {
+                self.invalidate_channel_index(clan_id);
+            }
+            if self.active_channel_id == Some(channel_id) {
+                let redirect = (!parent_id.is_zero()).then_some(parent_id);
+                self.active_channel_id = redirect;
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs:2321]: No navigate here, so the message list swaps to the parent while the route still renders the archived thread.

File: crates/mezon-store/src/channel.rs [hunk 4]
```diff
@@ -2109,11 +2400,37 @@ impl ChannelList {
     }
 }
 
+pub fn should_reactivate_after_send(mode: i32, channel: &Channel) -> bool {
+    should_reactivate_thread_after_send(mode, channel, channel.is_archived())
+}
+
+fn should_reactivate_thread_after_send(mode: i32, channel: &Channel, archived: bool) -> bool {
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs:2572]: Also reactivate when now - last_sent_timestamp > 7 days; React checks isArchived || !channel.active, not just active.

File: crates/mezon-store/src/channel.rs [hunk 5]
```diff
@@ -121,12 +124,21 @@ pub struct Channel {
...
 impl Channel {
     pub fn is_unread(&self) -> bool {
         self.badge_count > 0 || self.last_seen_timestamp < self.last_sent_timestamp
     }
+
+    pub fn is_archived(&self) -> bool {
+        self.active == CHANNEL_ACTIVE_ARCHIVED
+    }
+
+    pub fn visible_in_sidebar(&self) -> bool {
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs:151]: React keeps the open archived thread in the sidebar (no active filter). Confirm this divergence is intended.

File: crates/mezon-store/src/messages.rs [hunk 6]
```diff
@@ -4788,6 +4793,56 @@ impl MessagesStore {
...
+        mode: i32,
+        cx: &mut Context<Self>,
+    ) {
+        ChannelList::global(cx).update(cx, |list, cx| {
+            list.maybe_reactivate_after_send(channel_id, clan_id, mode, cx);
+        });
+    }
+}
+
+async fn ensure_archived_thread_reactivated(
```
Comment (huylexuan44) [inline crates/mezon-store/src/messages.rs:4850]: This path skips the reactivating guard that maybe_reactivate_after_send takes, so rapid sends fire duplicate calls.

File: crates/mezon-store/src/channel.rs [hunk 7]
```diff
@@ -1992,6 +2058,231 @@ impl ChannelList {
...
+
+    fn apply_channel_archive_event(
+        &mut self,
+        e: &mezon_proto::realtime::ChannelArchiveEvent,
+        cx: &mut Context<Self>,
+    ) {
+        let clan_id = ClanId(e.clan_id);
+        let channel_id = ChannelId(e.channel_id);
+        let parent_id = ChannelId(e.parent_id);
+        let is_archive = e.active == CHANNEL_ACTIVE_ARCHIVED;
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs:2299]: Add React's clan_id == 0 guard, otherwise the fallback below removes this channel from every clan cache.

File: crates/mezon-store/src/channel.rs [hunk 8]
```diff
@@ -2109,11 +2400,37 @@ impl ChannelList {
...
+}
+
+fn should_reactivate_thread_after_send(mode: i32, channel: &Channel, archived: bool) -> bool {
+    const STREAM_MODE_THREAD: i32 = 6;
+    mode == STREAM_MODE_THREAD
+        && (channel.channel_type == ChannelType::Thread || channel.parent_id.is_some())
+        && archived
+}
+
+fn sync_thread_active_status(existing: &mut Channel, active: i32, confirmed: bool) -> bool {
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs:2593]: This never goes ARCHIVED -> JOINED even when confirmed, which is why apply_channel_archive_event force-writes active.

File: crates/mezon-store/src/channel.rs [hunk 9]
```diff
@@ -1992,6 +2058,231 @@ impl ChannelList {
         }
     }
 
+    pub fn channel_active(&self, clan_id: ClanId, channel_id: ChannelId) -> Option<i32> {
+        self.channel(clan_id, channel_id).map(|ch| ch.active)
+    }
+
+    pub fn needs_reactivate_for_send(
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: Pure alias for prepare_archived_thread_for_reactivate. Drop one of the two.

File: crates/mezon-store/src/channel.rs [hunk 10]
```diff
@@ -1992,6 +2058,231 @@ impl ChannelList {
...
+                    Err(e) => {
+                        tracing::error!("active_archived_thread failed: {e}");
+                    }
+                }
+            });
+        })
+        .detach();
+    }
+
+    fn prepare_archived_thread_for_reactivate(
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: A needs_*/prepare_* predicate that mutates channel state is misleading. Split the check from the write-back.

File: crates/mezon-store/src/channel.rs [hunk 11]
```diff
@@ -2109,11 +2400,37 @@ impl ChannelList {
     }
 }
 
+pub fn should_reactivate_after_send(mode: i32, channel: &Channel) -> bool {
+    should_reactivate_thread_after_send(mode, channel, channel.is_archived())
+}
+
+fn should_reactivate_thread_after_send(mode: i32, channel: &Channel, archived: bool) -> bool {
+    const STREAM_MODE_THREAD: i32 = 6;
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: STREAM_MODE_THREAD already exists at mezon-store/src/messages.rs:479. Reuse it instead of redeclaring.

File: crates/mezon-client/src/transport_runtime.rs [hunk 12]
```diff
@@ -2956,3 +2996,25 @@ impl TransportClient {
             .map_err(|e| anyhow::anyhow!("transport task failed: {e}"))?
     }
 }
+
+fn is_request_timeout(err: &anyhow::Error) -> bool {
+    err.chain().any(|cause| {
+        let msg = cause.to_string();
+        msg.contains("Request timed out") || msg.contains("timed out")
```
Comment (huylexuan44) [inline crates/mezon-client/src/transport_runtime.rs]: contains("Request timed out") is subsumed by contains("timed out"), which also matches ping, TCP and HTTP timeouts.

File: crates/mezon-client/src/transport_runtime.rs [hunk 13]
```diff
@@ -1174,9 +1173,26 @@ impl TransportClient {
 
         runtime()
             .spawn(async move {
-                transport
-                    .list_thread_descs(channel_id, clan_id, THREAD_LIST_LIMIT, page, 0, None)
-                    .await
+                const MAX_RETRIES: u32 = 3;
```
Comment (huylexuan44) [inline crates/mezon-client/src/transport_runtime.rs]: Retry is unrelated to this PR, adds up to 7s latency, and the doc comment above was dropped. Split it out.

File: crates/mezon-client/src/app_api.rs [hunk 14]
```diff
@@ -2398,6 +2398,20 @@ impl AppApi {
             .await
     }
 
+    pub async fn active_archived_thread(&self, clan_id: i64, channel_id: i64) -> Result<()> {
+        self.transport
+            .active_archived_thread(clan_id, channel_id)
+            .await
+    }
+
+    pub async fn archive_channel(&self, clan_id: i64, channel_id: i64) -> Result<()> {
```
Comment (huylexuan44) [inline crates/mezon-client/src/app_api.rs]: archive_channel and leave_thread have no callers anywhere. Wire them up or drop them.

File: crates/mezon-ui/src/chat/layout.rs [hunk 15]
```diff
@@ -1092,6 +1094,31 @@ impl ChatLayout {
...
+            channel_id,
+            thread_id,
+        } = Router::global(cx).read(cx).route()
+        else {
+            return;
+        };
+        if self
+            .channel_list
+            .read(cx)
+            .channel_in_clan(clan_id, thread_id)
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/layout.rs]: No loaded-clan guard: thread deep links for an uncached clan get bounced to the parent. Gate on is_clan_cache_loaded.

File: crates/mezon-ui/src/sidebar/channel_sidebar/mod.rs [hunk 16]
```diff
@@ -317,7 +317,11 @@ impl ChannelSidebar {
                         collapsed,
                     });
                     if !collapsed {
-                        let ch_slice = &category.channels;
+                        let ch_slice: Vec<_> = category
+                            .channels
+                            .iter()
+                            .filter(|ch| ch.visible_in_sidebar())
```
Comment (huylexuan44) [inline crates/mezon-ui/src/sidebar/channel_sidebar/mod.rs:338]: Archived rows hidden here but still counted by clan_total_badge, leaving a lit clan badge with no row to clear.

File: crates/mezon-store/src/messages.rs [hunk 17]
```diff
@@ -3546,6 +3546,7 @@ impl MessagesStore {
                     let confirmed =
                         message_from_api(sent, AppConfig::try_global(cx), viewer_user_id(cx));
                     this.reconcile_temp(channel_id, temp_id, confirmed, cx);
+                    this.maybe_reactivate_archived_thread(channel_id, clan_id, mode, cx);
```
Comment (huylexuan44) [inline crates/mezon-store/src/messages.rs]: Duplicate unarchive: pre-send call at 3481 already ran and last_sent_timestamp is still stale here. React calls it once.

File: crates/mezon-store/src/threads.rs [hunk 18]
```diff
@@ -192,10 +196,70 @@ impl ThreadsStore {
...
+    fn apply_channel_archive(
+        &mut self,
+        ev: &realtime::ChannelArchiveEvent,
+        cx: &mut Context<Self>,
+    ) {
+        let channel_id = ev.channel_id.to_string();
+        let active = if ev.active == THREAD_STATUS_ARCHIVED {
+            THREAD_STATUS_ARCHIVED
+        } else {
+            THREAD_STATUS_JOINED
```
Comment (huylexuan44) [inline crates/mezon-store/src/threads.rs]: Flattens ACTIVE_PUBLIC/ACTIVE_PRIVATE to JOINED, moving unjoined threads into the Joined bucket. Use ev.status instead.

File: crates/mezon-store/src/channel.rs [hunk 19]
```diff
@@ -245,6 +259,7 @@ pub struct ChannelList {
     show_empty_categories: HashSet<ClanId>,
     channel_index: RefCell<ChannelLocationCache>,
     reset_generation: u64,
+    reactivating: HashSet<ChannelId>,
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs:278]: reset() clears every other per-session set but not this one, so a logout mid-request latches the id forever.

File: crates/mezon-store/src/channel.rs [hunk 20]
```diff
@@ -1995,6 +2063,219 @@ impl ChannelList {
...
+            let mut removed = false;
+            if let Some(cats) = self.cache.get_mut(&clan_id) {
+                removed = remove_channel(cats, channel_id);
+            } else {
+                for cats in self.cache.values_mut() {
+                    removed |= remove_channel(cats, channel_id);
+                }
+            }
+            if removed {
+                self.invalidate_channel_index(clan_id);
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: Else branch sweeps all clans but invalidates only clan_id. Use invalidate_channel_index_all like UserChannelRemoved does.

File: crates/mezon-store/src/channel.rs [hunk 21]
```diff
@@ -1995,6 +2063,219 @@ impl ChannelList {
...
+            .unwrap_or_default();
+        let name = label.filter(|s| !s.is_empty()).unwrap_or(existing_label);
+
+        if let Some(ch) = self.channel_mut(clan_id, channel_id) {
+            ch.active = CHANNEL_ACTIVE_JOINED;
+            if !name.is_empty() {
+                ch.name = name;
+            }
+        } else if let Some(parent_id) =
+            parent_id.or_else(|| self.active_channel_id.filter(|id| *id != channel_id))
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: Async fallback to active_channel_id re-parents the thread onto whatever channel is open when the response lands.

File: crates/mezon-ui/src/chat/layout.rs [hunk 22]
```diff
@@ -1842,15 +1869,35 @@ impl ChatLayout {
...
             return;
         };
-        self.thread_popover_handle.hide(cx);
-        self.clear_thread_search(cx);
+        self.dismiss_threads_popover(cx);
         let label = label.to_string();
         let parent = parent_id.parse::<ChannelId>().ok();
+        let (active, active_confirmed) = match ThreadsStore::global(cx)
+            .read(cx)
+            .thread_active(&channel_id.to_string())
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/layout.rs:1880]: thread_active returns THREAD_STATUS_* (0..3) but Channel.active is 0/1. Normalize to != 0 before storing.

File: crates/mezon-store/src/channel.rs [hunk 23]
```diff
@@ -1995,6 +2063,219 @@ impl ChannelList {
         }
     }
 
+    pub fn channel_active(&self, clan_id: ClanId, channel_id: ChannelId) -> Option<i32> {
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: channel_active has no callers. Wire it up or delete it.

File: crates/mezon-store/src/channel.rs [hunk 24]
```diff
@@ -2112,11 +2393,77 @@ impl ChannelList {
     }
 }
 
+pub fn should_reactivate_after_send(mode: i32, channel: &Channel) -> bool {
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: should_reactivate_after_send is only called from tests; production goes through reactivate_state_for_send.

File: crates/mezon-client/src/transport_runtime.rs [hunk 25]
```diff
@@ -1175,7 +1175,6 @@ impl TransportClient {
             .map_err(|e| anyhow::anyhow!("transport task failed: {e}"))?
     }
 
-    /// List threads for a parent channel.
     pub async fn list_thread_descs(
```
Comment (huylexuan44) [inline crates/mezon-client/src/transport_runtime.rs:1244]: Doc comment removal looks accidental. Restore "List threads for a parent channel."

File: crates/mezon-store/src/messages.rs [hunk 26]
```diff
@@ -4788,6 +4792,61 @@ impl MessagesStore {
...
+        mode: i32,
+        cx: &mut Context<Self>,
+    ) {
+        ChannelList::global(cx).update(cx, |list, cx| {
+            list.maybe_reactivate_after_send(channel_id, clan_id, mode, cx);
+        });
+    }
+}
+
+async fn ensure_archived_thread_reactivated(
```
Comment (huylexuan44) [inline crates/mezon-store/src/messages.rs:4850]: React handleThreadActivation also joins the sender to the thread when not a member. Intentionally out of scope?

File: crates/mezon-client/src/transport_runtime.rs [hunk 27]
```diff
@@ -3001,6 +3001,14 @@ impl TransportClient {
             .map_err(|e| anyhow::anyhow!("transport task failed: {e}"))?
     }
 
+    pub async fn active_archived_thread(&self, clan_id: i64, channel_id: i64) -> Result<()> {
```
Comment (huylexuan44) [inline crates/mezon-client/src/transport_runtime.rs]: Build is broken at this head: duplicate active_archived_thread - develop already added an identical copy at line 2567 (85c5854), delete this one.

File: crates/mezon-store/src/channel.rs (hunk 5 above)
Comment (huylexuan44) [reply crates/mezon-store/src/channel.rs:151]: Still open after the fixes: desktop hides the archived thread and redirects, while React keeps the open one visible in the sidebar. Please confirm this divergence is intended.

File: crates/mezon-store/src/messages.rs (hunk 26 above)
Comment (huylexuan44) [reply crates/mezon-store/src/messages.rs:4850]: Still unanswered: is joining the sender to the thread (React handleThreadActivation) intentionally out of scope?

**Comment by huylexuan44** [comment]
Re-reviewed at 023d6bc: 21 of 25 review comments are properly fixed (verified each one); with the duplicate below removed, mezon-store + mezon-client tests pass 881/881.
- Blocker: this head does not compile - duplicate active_archived_thread in transport_runtime.rs introduced by the last develop merge (inline comment).
- Leftover from the archive_channel/leave_thread cleanup: the MezonTransport methods themselves (transport.rs:5928, 5943) now have zero callers - worth deleting too.
- Two review questions are still unanswered (replied on their threads).
- Unrelated to this PR: clippy -D warnings currently fails on develop at stream.rs:446 (needless_bool_assign, from f8abba01) - needs a separate fix or CI stays red.
